lumigator/python/mzai/sdk/completions.py

```diff
@@ -0,0 +1,52 @@
+import json
+from http import HTTPMethod, HTTPStatus
+
+from schemas.completions import CompletionResponse
+
+from client import ApiClient
+
+
+class Completions:
+    COMPLETIONS_ROUTE = "completions"
+
+    def __init__(self, c: ApiClient):
+        self.__client = c
+        self.__cached_vendors = self.get_vendors()
+
+    def get_vendors(self) -> list[str]:
+        """Returns the list of supported external vendors."""
+        response = self.__client.get_response(self.COMPLETIONS_ROUTE)
+
+        if not response:
+            return []
+
+        # Update the cached vendors.
+        self.__cached_vendors = [str(vendor).lower() for vendor in response.json()]
+
+        return self.__cached_vendors
+
+    def get_completion(self, vendor: str, text: str) -> CompletionResponse | None:
+        """Returns completions from the specified vendor for given text (prompt)."""
+
+        # Sanitize the inputs.
+        vendor = vendor.lower()
+        text = text.strip()
+
+        # Attempt to validate vendors using the cache.
+        if vendor not in self.__cached_vendors:
+            raise ValueError(f"vendor '{vendor}' not supported")
```

Comment:
```suggestion
            raise ValueError(f"vendor '{vendor}' not found in cache, try reloading vendors")
```
...or something like it

Reply:
Good point! I'll update it now.